src/Control/Email/MailerSubscriber.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+namespace SilverStripe\Control\Email;
+
+use InvalidArgumentException;
+use SilverStripe\Control\HTTP;
+use SilverStripe\Core\Extensible;
+use SilverStripe\Core\Injector\Injectable;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Mailer\Messenger\SendEmailMessage;
+use Symfony\Component\Messenger\Event\SendMessageToTransportsEvent;
+use Symfony\Component\Mime\Address;
+
+/**
+ * This subscriber is registered in BaseKernel->bootEmail()
+ *
+ * See https://symfony.com/doc/current/mailer.html#mailer-events for further info
+ */
+class MailerSubscriber implements EventSubscriberInterface
+{
+    use Injectable;
+    use Extensible;
+
+    public static function getSubscribedEvents()
+    {
+        return [
+            SendMessageToTransportsEvent::class => 'onSendMessageToTransports',
+        ];
+    }
+
+    public function onSendMessageToTransports(SendMessageToTransportsEvent $event): void
+    {
+        $envelope = $event->getEnvelope();
+        /** @var SendEmailMessage $sendEmailMessage */
+        $sendEmailMessage = $envelope->getMessage();
+        $email = $sendEmailMessage->getMessage();
```

Comment:
```suggestion
        $email = $event->getEnvelope()->getMessage()->getMessage();
```
We don't need those intermediate variables. They're not doing anything.

Reply:
Has been changed to work with MessageEvent::class